aiidalab_widgets_base/utils/__init__.py

```diff
@@ -88,6 +93,21 @@ def string_range_to_list(strng, shift=-1):
     return singles, True
 
 
+def get_formula(data_node):
+    """A wrapper for getting a molecular formula out of the AiiDA Data node"""
+    if isinstance(data_node, TrajectoryData):
+        # TrajectoryData can only hold structures with the same chemical formula,
+        # so this approach is sound.
```

Comment:
Are you sure? If one runs Grand Canonical MC the number of particles does change.

Reply:
I tried creating TrajectoryData from two different StructureData nodes with different chemical composition and got an exception, but feel free to verify. (tested with AiiDA 1.6 in AiiDAlab image)

Reply:
Ok, if that is the constraint they have, maybe we keep it like this for the moment. But please mention this in the comment above. Just to avoid that people will have to re-discover this in the future. Thanks!

Reply:
To me the comment says that already. :-) Do you have a specific wording in mind?

Reply:
You are right, sorry, all good.